python/lsst/obs/pfs/constructFiberFlatTask.py

```diff
@@ -143,7 +138,10 @@ def combine(self, cache, struct):
         xOffsets = []
         for expRef in dataRefList:
             exposure = expRef.get('postISRCCD')
-            xOffsets.append(exposure.getMetadata().get('sim.slit.xoffset'))
+            try:
+                xOffsets.append(exposure.getMetadata().get('sim.slit.xoffset'))
```

Comment:
You cannot assume that you're only processing simulations!  

Also, you may not read values you need from metadata, ever.  If you need a value it must be moved into an object (e.g. we have Wcs and Calib rather than reading the keys from the metadata).